util.py

```diff
@@ -170,102 +182,22 @@ def _entity_metadata(dataset, properties, metadata={}, parent_key=''):
     return metadata
 
 
-# TODO: create job should be use-able from bootstrap.py with only database information
-def create_job(dbc, datainfo):
-    fileinfo = datainfo['fileinfo']
-    db = dbc.database
-    type_ = fileinfo['filetype']
-    kinds_ = fileinfo['datatypes']
-    state_ = ['orig'] # dataset.nims_file_state ### WHAT IS THIS AND WHY DO WE CARE?
-    app = None
-    # TODO: check if there are 'default apps' set for this project/session/acquisition
-    acquisition = db.acquisitions.find_one({'uid': datainfo['acquisition_id']})
-    session = db.sessions.find_one({'_id': acquisition.get('session')})
-    project = db.projects.find_one({'_id': session.get('project')})
-    aid = acquisition.get('_id')
-
-    # XXX: if an input kinds = None, then that job is meant to work on any file kinds
-    app = db.apps.find_one({
-        '$or': [
-            {'inputs': {'$elemMatch': {'type': type_, 'state': state_, 'kinds': kinds_}}, 'default': True},
-            {'inputs': {'$elemMatch': {'type': type_, 'state': state_, 'kinds': None}}, 'default': True},
-        ],
-    })
-    # TODO: this has to move...
-    # force acquisition dicom file to be marked as 'optional = True'
-    db.acquisitions.find_and_modify(
-        {'uid': datainfo['acquisition_id'], 'files.type': 'dicom'},
-        {'$set': {'files.$.optional': True}},
-        )
-
-    if not app:
-        log.info('no app for type=%s, state=%s, kinds=%s, default=True. no job created.' % (type_, state_, kinds_))
+def identical_content(filepath1, digest1, filepath2, digest2):
+    if zipfile.is_zipfile(filepath1) and zipfile.is_zipfile(filepath2):
+        with zipfile.ZipFile(filepath1) as zf1, zipfile.ZipFile(filepath2) as zf2:
+            zf1_infolist = sorted(zf1.infolist(), key=lambda zi: zi.filename)
+            zf2_infolist = sorted(zf2.infolist(), key=lambda zi: zi.filename)
+            if zf1.comment != zf2.comment:
+                return False
+            if len(zf1_infolist) != len(zf2_infolist):
+                return False
+            for zii, zij in zip(zf1_infolist, zf2_infolist):
+                if zii.CRC != zij.CRC:
+                    return False
+            else:
+                return True
```

Comment:
I'm very worried about going down the rabbit hole of container-agnostic content comparison that digs into tars & zips to see what's inside.

Our execution library has nascent consistent-filesystem-hashing functionality that tries to formalize what this function attempts, and it turns out to be a deceptively-complex problem requiring a lot of thought, edge case handling, and test coverage.

I would much rather the api not open up filesystem containers until & unless we sit down and design + test that feature wholesale. This would have the advantage of the engine & the server agreeing on how filesystems are described.